u8timeseries/timeseries.py

```diff
@@ -135,49 +230,98 @@ def _slice_not_none(s: Optional[pd.Series]) -> Optional[pd.Series]:
 
     def slice_n_points_after(self, start_ts: pd.Timestamp, n: int) -> 'TimeSeries':
         """
-        Returns a new time series, starting later than [start_ts] (inclusive) and having (at most) [n] points
-        :param start_ts:
-        :param n:
-        :return:
+        Returns a new TimeSeries, starting later than [start_ts] (included) and having (at most) [n] points.
+
+        The timestamp may not be in the time series. If it is, it will be included in the new TimeSeries.
+
+        :param start_ts: The timestamp that indicates the splitting time.
+        :param n: The maximal length of the new TimeSeries.
+        :return: A new TimeSeries, with length at most [n] and indices greater or equal than [start_ts].
         """
-        end_ts: pd.Timestamp = start_ts + (n-1) * self.freq()  # (n-1) because slice() is inclusive on both sides
+
+        assert n >= 0, 'n should be a positive integer.'
+
+        if (start_ts < self.start_time()) or (start_ts > self.end_time()):
```

Comment:
How about having a function `_raise_if_not_within(ts: pd.Timestamp) -> bool` in order to avoid repeating this code?